### ISSUE TITLE
[Help needed] Dynamically set an element's position within a parent

### ISSUE THREAD

**Issue by spaceemotion** [nifty-gui/nifty-gui#251]
Hey, I got some kind of a problem here; I want to dynamically set the position of an element within a panel.

So far I got it working by getting the parent's position + the wanted position, but this seems to by some kind of hacky solution if I not update the value each time the layout changes.

Is there any way to dynamically set the X / Y coordinate of an element WITHIN its parent? The "setConstraint" methods do not work at all…

I also got a weird problem when I want to have two images and a text field above each other (within a panel with "center" child-layout); The two images are visible, but not the text. If I remove the second image, the text re-appears. Can it be that, although the image's bounds are 90 x 8 pixels, it pushes the text out of the panel? If that would be the case though, would't the panel get resized (I never set the width and height, it takes the size of the first - the biggest - image).

Thanks in advance!

Edit: Re-reading the manual, this is only allowed when using absolute layouts ....

**Comment by relu91** [comment]
> Edit: Re-reading the manual, this is only allowed when using absolute layouts ....

Exactly, about the other problem could you provide a simple xml to test  that bug? I've tried in nifty-editor and the text seems fine, but I could possible misunderstand your gui configuration :)

**Comment by spaceemotion** [comment]
Sorry, that I reply that late; here is the XML excerpt for the custom controls:

``` xml
    <!-- Image Progress bar -->
    <controlDefinition style="catharos-image-progress" name="image-progress"
                       controller="xxxxxxx.gui.controllers.ImageProgressController">
        <panel childLayout="center">
            <image id="#image" filename="$image" />
            <text id="#text" style="#label" text="$label" />
        </panel>
    </controlDefinition>

    <!-- Image Progress bar -->
    <controlDefinition style="catharos-image-progress2" name="image-progress2"
                       controller="xxxxxxx.gui.controllers.ImageProgress2Controller">
        <panel childLayout="absolute-inside" width="d" height="d">
            <image id="#image" filename="$image" align="center" valign="center" />
            <image id="#overlay" filename="$overlay" />
            <text id="#text" style="#label" text="$label" align="center" valign="center" />
        </panel>
    </controlDefinition>
```

The first one works perfectly fine, the second one strangely doesn't show the text...

**Comment by relu91** [comment]
Have you tried with different childLayouts ? I mean change the panel childLayout to "center" ?

**Comment by spaceemotion** [comment]
yes I did, they won't move at all then …

**Comment by relu91** [comment]
Oh well , I didn't read your first post again sorry :S . So does the text disappear for every layout? or only with the center one ?
